src/lib.rs

```diff
@@ -628,17 +674,26 @@ pub enum Expression {
         accept: Handle<Expression>,
         reject: Handle<Expression>,
     },
-    /// Call an intrinsic function.
-    Intrinsic {
-        fun: IntrinsicFunction,
+    /// Compute the derivative on an axis.
```

Comment:
I wonder if these need special handling in the IR itself? If we were able to reuse `Call` with something like `FunctionOrigin::Intrinsic` then it seems like we'd be able to simplify the writers a little

Reply:
I thought derivatives are "special" because they depend on the hidden state of the pipeline - the quad-based rasterization. Using them also requires uniform control flow. So that makes them different enough to differentiate from other functions that are mostly math.

Reply:
Ah yeah I guess derivatives are probably a bit different. I was mostly wondering about whether we could collapse `Math` and `Relational` into `Call` somehow

Reply:
Yeah, we could totally do that. I'm not feeling strongly, just got inspired by the fact the relational stuff is separate in both current WGSL spec draft and the MSL "specification".